wiki-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java

```diff
@@ -91,30 +94,68 @@ public Response getNote(@Context UriInfo uriInfo,
+  public Response getNoteById(@ApiParam(value = "Note id", required = true) @PathParam("noteId") String noteId) {
+    try {
+      Identity identity = ConversationState.getCurrent().getIdentity();
+      Page note = noteBookService.getPageById(noteId);
+      if (!noteBookService.hasPermissionOnPage(note, PermissionType.VIEWPAGE, identity)) {
+        return Response.status(Response.Status.FORBIDDEN).build();
+      }
+      if (note == null) {
+        return Response.status(Response.Status.NOT_FOUND).build();
+      }
+      note.setContent(WikiHTMLSanitizer.markupSanitize(note.getContent()));
+      note.setBreadcrumb(noteBookService.getBreadcumb(note.getWikiType(),note.getWikiOwner(),noteId));
+      return Response.ok(note).build();
+    } catch (Exception e) {
+      log.error("Can't get note {}",noteId, e);
       return Response.serverError().entity(e.getMessage()).build();
     }
   }
 
-  /**
-   * Create new note
-   *
-   * @param note Note.
-   * @return {@link Response} with status HTTPStatus.ACCEPTED if saving process is
-   *         performed successfully with status HTTPStatus.INTERNAL_ERROR if there
-   *         is any unknown error in the saving process
-   */
   @POST
   @Path("/note")
   @RolesAllowed("users")
-  public Response createNote(@ApiParam(value = "task object to be updated", required = true) Page note) {
+  @ApiOperation(value = "Add a new note", httpMethod = "POST", response = Response.class, notes = "This adds a new note.")
+  @ApiResponses(value = { @ApiResponse(code = 200, message = "Request fulfilled"),
+          @ApiResponse(code = 400, message = "Invalid query input"), @ApiResponse(code = 403, message = "Unauthorized operation"),
+          @ApiResponse(code = 404, message = "Resource not found") })
+  public Response createNote(@ApiParam(value = "note object to be created", required = true) Page note) {
     if (note == null) {
       return Response.status(Response.Status.BAD_REQUEST).build();
     }
+    String noteBookType=note.getWikiType();
+    String noteBookOwner=note.getWikiOwner();
     try {
       Identity identity = ConversationState.getCurrent().getIdentity();
+      if(StringUtils.isNotEmpty(note.getParentPageId())){
+        Page note_=noteBookService.getPageById(note.getParentPageId());
+        if(note_!=null){
+          noteBookType=note_.getWikiType();
+          noteBookOwner=note_.getWikiOwner();
+          note.setParentPageName(note_.getName());
+        }else{
+          return Response.status(Response.Status.BAD_REQUEST).build();
+        }
+      }
+      if(StringUtils.isEmpty(noteBookType) || StringUtils.isEmpty(noteBookOwner)){
+        return Response.status(Response.Status.BAD_REQUEST).build();
+      }
       /* TODO: check noteBook permissions */
-      Wiki noteBook = noteBookService.getWikiByTypeAndOwner(note.getWikiType(), note.getWikiOwner());
+      Wiki noteBook = noteBookService.getWikiByTypeAndOwner(noteBookType, noteBookOwner);
       if (noteBook == null) {
```

Comment:
@mkrout shouldn’t we check that user has permission to write in this nodebook before creating the page?

Reply:
In this case the check of permission is already done in the service layer (old wiki implementation): noteBookService.getWikiByTypeAndOwner() return the note only if the current user has the view permission (https://github.com/Meeds-io/notes/blob/develop/wiki-service/src/main/java/org/exoplatform/wiki/service/impl/WikiServiceImpl.java#L488) so even if the page exists if the user don't have the view permission the method will return null.

Reply:
I see but I find it weid that some endpoints take tha responsability tu permission check and others no. 
Don’t we have a guideline to always start with permission checks at the endpoint layer?

Reply:
You're right, and that what I did in all other calls using getPagebyId() since the permission check is not done on the service layer. for the calls using getWikiByTypeAndOwner() as I said the check is done in the service and it's not recommended but that's how the old implementation was done and we can't change it now since it's used in the old wiki application, We will change that when we remove the old portlet.

Reply:
What problem is there in having permission check here and duplicated in the service layer? This way, you don’t leave any debt behind in case we’d decide to build a new service layer.